### PR TITLE
Web meric - Sign in with google + Legal Documents + address backend connection

### PR DESCRIPTION
**Pull request by merijjeyn**
- Sign in with google button on sign in and sign up
- Added Conditions of Use document and connected legal documents to sign up and checkout pages
- Customer addresses page backend connection

### CODE REVIEW

**Review by eylulyalcinkaya — APPROVED** [review]

**Comment by burakcuhadar** [comment]
I get the following error on the sign in page:
TypeError: Cannot read property 'signin2' of undefined

File: web/src/pages/CustomerProfile.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 
 import ProfileInfo from '../components/customer-profile/ProfileInfo'
 import UpdatePassword from '../components/customer-profile/UpdatePassword'
 
 import './CustomerProfile.css'
 
-const CustomerProfile = () => {
-
+const CustomerProfile = ({ isGoogleUser }) => {
+	console.log("Is Google User? ", isGoogleUser)
```
Comment (BCGpp) [inline web/src/pages/CustomerProfile.js:10]: console.log not removed.

**Review by BCGpp — COMMENTED** [review]
This is not reflected on the wiki, but login requests (using /login endpoint) normally has a 200 response with a data field like this: { "userId": <userId>, "status": <accountStatus> }
where accountStatus is "verified", "not-verified", or "banned"
Presumably (ideally), google sign-in requests (using /google-signin) should produce a response in the same format. As a result, the onGoogleSignIn function (in SignIn.js) should do the appropriate check for banned users, i.e. looking at response.status instead of response.banned.
Ofc, disregard this if the backend response ends up being in a different format. Right now, it looks like the backend isn't implemented & you're using a constant response.

**Comment by merijjeyn** [comment]
> This is not reflected on the wiki, but login requests (using /login endpoint) normally has a 200 response with a data field like this: { "userId": <userId>, "status": <accountStatus> }
> where accountStatus is "verified", "not-verified", or "banned"
> Presumably (ideally), google sign-in requests (using /google-signin) should produce a response in the same format. As a result, the onGoogleSignIn function (in SignIn.js) should do the appropriate check for banned users, i.e. looking at response.status instead of response.banned.
> Ofc, disregard this if the backend response ends up being in a different format. Right now, it looks like the backend isn't implemented & you're using a constant response.

Google sign in is handled on a different function called onGoogleSignIn so that wont be an issue

**Comment by merijjeyn** [comment]
> I get the following error on the sign in page:
> TypeError: Cannot read property 'signin2' of undefined
> 
> To reproduce you can try to refresh the page several times, it does not always occur.

Should be fixed

**Pull request merged**